### ISSUE TITLE
Hoping for the full installation of the code

### ISSUE THREAD

**Issue by Chen-yuiyui** [jimmy15923/unsup_point_coseg#4]
Sorry, thank you for your work. But I found a few mistakes.

1. s3dis_h5_07042019 should be the correct dataset which should be preprocessed.The link you provided seems like is wrong,
2. What is the .pcd fold ?  I see the code from dataset fold said "#from .pcd import normalize_data, center_data, ori_rotate_pointcloud",but where is it?How can I find them?
3. It seems like you use the pointnet2_pytorch, I copy the pointnet2_ops.lib from [https://github.com/erikwijmans/Pointnet2_PyTorch](url) , ,change the code 
 _from pointnet2_ops_lib.pointnet2_ops.pointnet2_utils import grouping_operation as group_point_
 and build the kernels, seems the pretrain.py could work correctly. So can you provide the full installation readme?

**Comment by jimmy15923** [comment]
Hi! Thanks for being interested in our work.

1. Thanks for pointing out the wrong link to the used dataset. The dataset is obtained in the [jsis3d repo](https://github.com/pqhieu/jsis3d/issues/20#issuecomment-589619628)
2. The normalization of point cloud code is from [ScanObjetNN repo](https://github.com/hkust-vgd/scanobjectnn/blob/fe60aeade9ceb8882bc3f1bc40612e65469d7e77/data_utils.py#L133)
3. The Pointnet2_PyTorch used in SampleNet seems a little bit different from the official latest repo. That's why I suggest using the docker file to avoid the environment issues. Please see the [SampleNet installation](https://github.com/itailang/SampleNet/tree/master/registration#inside-the-docker-container) for more details

**Comment by Chen-yuiyui** [comment]
`def __getitem__(self, item):
        pointcloud = self.data[item]

        label = self.label[item]
        if self.partition == 'train':
            pointcloud = translate_pointcloud(pointcloud)
            pointcloud = ori_jitter_pointcloud(pointcloud)
            pointcloud = ori_rotate_pointcloud(pointcloud)
            np.random.shuffle(pointcloud)
            pointcloud = pointcloud[:self.num_points]

        pointcloud = center_data(pointcloud)
        pointcloud = normalize_data(pointcloud)
        return pointcloud, label`
Thanks, I find the center_data and normalize_data from your link , but I can't find the  translate_pointcloud\ori_jitter_pointcloud\ori_rotate_pointcloud functions,where can I get them?
Thank you.

**Comment by jimmy15923** [comment]
Hi! 
Just notice that I neglect the `pcd.py` file. Already uploaded in [dataset/pcd.py](https://github.com/jimmy15923/unsup_point_coseg/blob/main/datasets/pcd.py). Thanks for pointing out this issue.
Btw, those process functions are from the [DGCNN repo](https://github.com/AnTao97/dgcnn.pytorch/blob/master/data.py).

**Comment by Chen-yuiyui** [comment]
I‘m sorry to bother you again.But could you please upload the config.yaml file such as scanobj.yaml or s3dis.yaml?

**Comment by jimmy15923** [comment]
Hi @Chen-yuiyui 

Already uploaded!
